src/Sylius/Component/Core/Cart/Context/ShopBasedCartContext.php

```diff
@@ -26,7 +26,7 @@
 use Sylius\Component\Order\Model\OrderInterface as BaseOrderInterface;
 use Webmozart\Assert\Assert;
 
-final class ShopBasedCartContext implements CartContextInterface
+final class ShopBasedCartContext implements ShopBasedCartContextInterface
```

Comment:
```suggestion
final class ShopBasedCartContext implements CartContextInterface, ResettingCartContextInterface
```

Explanation below 👇🏼.